torch/csrc/jit/interpreter.cpp

```diff
@@ -218,6 +230,31 @@ Operation getOperation(jit::Node *node) {
     return [](const list_of_retainable & inputs, list_of_retainable & outputs) {
       outputs.push_back(toRetainableSteal(at::Tensor()));
     };
+  IR_ELSEIF(ReplaceIfUndef)
+    return [](const list_of_retainable & inputs, list_of_retainable & outputs) {
+      auto result = inputs[0];
+      //TODO: refcounting stuff here is ugly but TensorTemporary is not
+      //present. Consider whether we
+      // 1. expose tensor temporary here
+      // 2. keep as is
+      // 3. remove all of this retainable stuff anyway since the new
+      // execution paths do not need handle types.
+      // Note that list_of_retainable is painful because it is yet another
+      // list of pointers that require needless copies.
+      if(result == at::UndefinedTensor::singleton()) {
+        result = inputs[1];
+      }
+      result->retain();
+      outputs.push_back(result);
+    };
+  IR_ELSEIF(GraphExecutor)
+    GraphExecutor executor(value->g(kSubgraph));
+    return [=](const list_of_retainable & inputs, list_of_retainable & outputs) mutable {
+      autograd::profiler::RecordFunction record("GraphExecutor");
+      tensor_list tinputs = unsafeToTensorListShare(inputs);
+      tensor_list toutputs = executor.run(tinputs);
```

Comment:
This makes me a bit nervous. I understand that we create temporary tensors that don't increase the refcount in `tinputs` and give a copy to `executor.run`. Now, do these copies retain? Do they decref when they are freed?

Reply:
Again, we need to redo this stuff anyway, but this is exactly the same code as FusionGroup.